event_sourcery/event_store/factory.py

```diff
@@ -1,49 +1,47 @@
 import abc
-from functools import partial
-from typing import Any, Callable, ContextManager, Iterator
+from typing import ContextManager, Iterator
 
 from typing_extensions import Self
 
-from event_sourcery.event_store.event import Event, EventRegistry, RawEvent, Serde
+from event_sourcery.event_store import subscription
+from event_sourcery.event_store.event import EventRegistry, RawEvent, Serde
 from event_sourcery.event_store.event_store import EventStore
 from event_sourcery.event_store.interfaces import (
     OutboxFiltererStrategy,
     OutboxStorageStrategy,
 )
+from event_sourcery.event_store.outbox import Outbox
 
 
 def no_filter(entry: RawEvent) -> bool:
     return True
 
 
 class NoOutboxStorageStrategy(OutboxStorageStrategy):
-    def put_into_outbox(self, events: list[RawEvent]) -> None:
-        pass
-
     def outbox_entries(self, limit: int) -> Iterator[ContextManager[RawEvent]]:
         return iter([])
 
 
-class EventStoreFactory(abc.ABC):
-    serde = Serde(Event.__registry__)
-    build: Callable[..., EventStore] = partial(
-        EventStore,
-        outbox_storage_strategy=NoOutboxStorageStrategy(),
-        serde=serde,
-    )
+class Engine:
+    serde: Serde
+    event_store: EventStore
+    outbox: Outbox
+    subscriber: subscription.Positioner
 
-    def _configure(self, **kwargs: Any) -> Self:
-        self.build = partial(self.build, **kwargs)
-        return self
 
+class EventStoreFactory(abc.ABC):
+    @abc.abstractmethod
+    def build(self) -> Engine:
```

Comment:
I'm not feeling it fully. Why not have separate methods for `build_event_store`, `build_outbox` etc?

Reply:
It's like committing to configuration.

For some reason I want it to be resilient for edge cases like building an event store and changing config, building outbox.